REGRESSION (281429@main): [ iOS macOS Debug ] ASSERTION FAILED: requestData->requestMap.contains(path) result of TestWebKitAPI.WKNavigation.HTTPSOnlyInitialLoad to be a consistent crash

sysrqb opened the pull request:
#### 80c1e6d05e4679c08e3a6e16b61ca8059bd39245
<pre>
REGRESSION (281429@main): [ iOS macOS Debug ] ASSERTION FAILED: requestData-&gt;requestMap.contains(path) result of TestWebKitAPI.WKNavigation.HTTPSOnlyInitialLoad to be a consistent crash
<a href="https://bugs.webkit.org/show_bug.cgi?id=277568">https://bugs.webkit.org/show_bug.cgi?id=277568</a>
<a href="https://rdar.apple.com/133089106">rdar://133089106</a>

Reviewed by Alex Christensen.

Define the /notsecure path in the http server configuration.

This change also fixes <a href="https://rdar.apple.com/133425387">rdar://133425387</a> by deleting the HTTP proxy
configuration and directly loading the http page from the server. This change
lets us simply define the path for the http resource, instead of specifying the
origin.

* Tools/TestWebKitAPI/Tests/WebKitCocoa/Navigation.mm:
(TEST(WKNavigation, HTTPSFirstLocalHostIPAddress)):
(TEST(WKNavigation, HTTPSOnlyInitialLoad)):

Canonical link: <a href="https://commits.webkit.org/282747@main">https://commits.webkit.org/282747@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f00dcd91b2191764ab9315c33a035ddd69613978

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/63321 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/42677 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/15918 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/67342 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/13929 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/65441 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/50365 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/14209 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/51012 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/9626 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/66390 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/39622 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/54834 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/31693 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/36305 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/12183 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/12801 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/57844 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/12511 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/69038 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/7268 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/12114 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/58321 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/7299 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/54904 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/58552 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/6056 "Passed tests") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/9706 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/38498 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/39577 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/40689 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/39320 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/Navigation.mm:
```diff
@@ -1714,14 +1702,14 @@ HTTPServer httpServer({
     HTTPServer httpsServer({
-        { "/secure"_s, { { }, "secure page"_s } }
+        { "/secure"_s, { { }, "secure page"_s } },
+        { "/notsecure"_s, { { }, "notsecure page upgraded to secure page"_s } }
```
Could we rename this path to /initially_not_secure or something to indicate that it is ok that it becomes secure?

sysrqb committed:
REGRESSION (281429@main): [ iOS macOS Debug ] ASSERTION FAILED: requestData->requestMap.contains(path) result of TestWebKitAPI.WKNavigation.HTTPSOnlyInitialLoad to be a consistent crash
https://bugs.webkit.org/show_bug.cgi?id=277568
rdar://133089106

Reviewed by Alex Christensen.

Define the /notsecure path in the http server configuration.

This change also fixes rdar://133425387 by deleting the HTTP proxy
configuration and directly loading the http page from the server. This change
lets us simply define the path for the http resource, instead of specifying the
origin.

* Tools/TestWebKitAPI/Tests/WebKitCocoa/Navigation.mm:
(TEST(WKNavigation, HTTPSFirstLocalHostIPAddress)):
(TEST(WKNavigation, HTTPSOnlyInitialLoad)):

Canonical link: https://commits.webkit.org/282747@main
